### PR TITLE
FFWS-102: Added date range filter

### PR DESCRIPTION
**Pull request by BogRad97**

### CODE REVIEW

File: src/fleming-theme/page-knowledge-resources.php [hunk 1]
```diff
@@ -21,12 +21,30 @@ function fleming_get_content() {
         'nav' => get_nav_builder()->withMenuRoute('knowledge')->build()
     );
 
+    $all_months = array('January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December');
     $type = !empty($_GET["type"]) && $_GET["type"] !== 'event' ? get_page_by_path($_GET["type"], 'OBJECT', 'publication_types') :
         (!empty($_GET["type"]) ? $_GET["type"] : null);
     $country = !empty($_GET["country"]) ? get_page_by_path($_GET["country"], 'OBJECT', 'countries') : null;
     $page_number = !empty($_GET["page_number"]) ? $_GET["page_number"] : 1;
+    $start_month = !empty($_GET["start_month"]) ? $_GET["start_month"] : null;
+    $end_month = !empty($_GET["end_month"]) ? $_GET["end_month"] : null;
```
Comment (TheMattBarnfield) [inline src/fleming-theme/page-knowledge-resources.php:30]: we have a lot of lines with this ugly pattern, perhaps we should extract to a method? Something like `getOrDefault`?

File: src/fleming-theme/page-knowledge-resources.php [hunk 2]
```diff
@@ -21,12 +21,30 @@ function fleming_get_content() {
...
-    $publications = get_knowledge_and_resources_publications_filtered_by_type_and_country($type, $country, $page_number, 10);
+    $start_month_idx = array_search($start_month, $all_months);
+    if ($start_month_idx != null) {
+        $start_month_idx = $start_month_idx + 1;
+    }
+    $end_month_idx = array_search($end_month, $all_months);
+    if ($end_month_idx == 11) {
+        $end_month_idx = 1;
+        $end_year = $end_year + 1;
+    }
```
Comment (TheMattBarnfield) [inline src/fleming-theme/page-knowledge-resources.php:42]: I'm a little confused by the intended behaviour here. After this the `end_month_idx` can be `0`-`10` of the same year or `1` of the next year?
What do we want this to do?

File: src/fleming-theme/templates/page-elements/knowledge-and-resources-filters-with-results.html [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        if (startRangeSet !== 1 && endRangeSet !== 1)
+        {
+            $('#filter-search-form').submit();
+        }
+    }
+
+    $(function () {
+        var currentPage = 1;
+        var maxPageNumber = {{query_result.max_page_number}};
+        var loading = false;
```
Comment (TheMattBarnfield) [inline src/fleming-theme/templates/page-elements/knowledge-and-resources-filters-with-results.html:24]: we're mixing `let` and `var` here- ideally we'd only use one (By default `let`, but `var` if we need the browser support)

**Review by TheMattBarnfield — COMMENTED** [review]
Thanks Bogdan- just want to confirm the end date behaviour before this gets merged

File: src/fleming-theme/page-knowledge-resources.php (hunk 2 above)
Comment (BogRad97) [reply src/fleming-theme/page-knowledge-resources.php:42]: If it's the last month of the year, then I want the date to be the first day of the next year

Comment (BogRad97) [reply src/fleming-theme/page-knowledge-resources.php:42]: Although I now realize I should've set the end_month_idx to 0, and increment it if it's not null.

File: src/fleming-theme/page-knowledge-resources.php [hunk 4]
```diff
@@ -40,9 +65,15 @@ function fleming_get_content() {
 
     $fleming_content["type_query"] = !empty($_GET["type"]) ? $_GET["type"] : null;
     $fleming_content["country_query"] = !empty($_GET["country"]) ? $_GET["country"] : null;
+    $fleming_content["start_month_query"] = !empty($_GET["start_month"]) ? $_GET["start_month"] : null;
+    $fleming_content["start_year_query"] = !empty($_GET["start_year"]) ? $_GET["start_year"] : null;
+    $fleming_content["end_month_query"] = !empty($_GET["end_month"]) ? $_GET["end_month"] : null;
+    $fleming_content["end_year_query"] = !empty($_GET["end_year"]) ? $_GET["end_year"] : null;
```
Comment (TheMattBarnfield) [inline src/fleming-theme/page-knowledge-resources.php:71]: I think you can use `get_or_default` here too

**Review by TheMattBarnfield — APPROVED** [review]
Cheers Bogdan, looks good!

**Pull request merged**